argon2/src/memory.rs

```diff
@@ -0,0 +1,162 @@
+//! Views into Argon2 memory that can be processed in parallel.
+//!
+//! This module implements, with a combination of compile-time borrowing and runtime checking, the
+//! cooperative contract described in section 3.4 (Indexing) of RFC 9106:
+//!
+//! > To enable parallel block computation, we further partition the memory matrix into SL = 4
+//! > vertical slices. The intersection of a slice and a lane is called a segment, which has a
+//! > length of q/SL. Segments of the same slice can be computed in parallel and do not reference
+//! > blocks from each other. All other blocks can be referenced.
+
+#![warn(
+    clippy::undocumented_unsafe_blocks,
+    clippy::missing_safety_doc,
+    unsafe_op_in_unsafe_fn
```

Comment:
I don't see any `unsafe` usage in `unsafe fn`? The only `unsafe fn` I see is `SegmentView::new`

Reply:
You're right, since adding the lint I simplified the implementation significantly. Might still be worthwhile to keep the lint though.

Reply:
Actually, since the crate switched to Rust 2024 (#563), the lint is warn by default anyways. So I'll remove it.

And I missed your edit before my previous reply. In the first version of this PR there were unsafe ops in unsafe functions, and I think it's good to have a unsafe block in those cases. Rust 2024 had already made the lint warn by default, but argon2 was still on Rust 2021.